examples/pcr/policy.c

```diff
@@ -0,0 +1,245 @@
+/* policy.c
+ *
+ * Copyright (C) 2006-2022 wolfSSL Inc.
+ *
+ * This file is part of wolfTPM.
+ *
+ * wolfTPM is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * wolfTPM is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+
+/* This is a helper tool for setting policies on a TPM 2.0 PCR */
+
+#include <wolftpm/tpm2_wrap.h>
+
+#ifndef WOLFTPM2_NO_WRAPPER
+
+#ifndef WOLFTPM2_NO_WOLFCRYPT
+#include <wolfssl/wolfcrypt/hash.h>
+#endif
+
+#include "hal/tpm_io.h"
+#include <examples/pcr/pcr.h>
+#include <examples/tpm_test.h>
+
+#include <stdio.h>
+
+static char HexCharToByte(char ch)
+{
+    char ret = (char)ch;
+    if (ret >= '0' && ret <= '9')
+        ret -= '0';
+    else if (ret >= 'A' && ret <= 'F')
+        ret -= 'A' - 10;
+    else if (ret >= 'a' && ret <= 'f')
+        ret -= 'a' - 10;
+    else
+        ret = -1; /* error case - return code must be signed */
+    return ret;
+}
+static int HexToByte(const char *hex, unsigned char *output, unsigned long sz)
+{
+    word32 i;
+    for (i = 0; i < sz; i++) {
+        char ch1, ch2;
+        ch1 = HexCharToByte(hex[i * 2]);
+        ch2 = HexCharToByte(hex[i * 2 + 1]);
+        if ((ch1 < 0) || (ch2 < 0)) {
```

Comment:
```
examples/pcr/policy.c: In function ‘HexToByte’:
examples/pcr/policy.c:58:18: error: comparison is always false due to limited range of data type [-Werror=type-limits]
   58 |         if ((ch1 < 0) || (ch2 < 0)) {
      |                  ^
examples/pcr/policy.c:58:31: error: comparison is always false due to limited range of data type [-Werror=type-limits]
   58 |         if ((ch1 < 0) || (ch2 < 0)) {
      |                               ^
```

Reply:
how did you get this error? char is a signed type and a negative char should be possible

Reply:
Cross-compiling for an aarch64 target:

```
./configure --host=aarch64 CC="aarch64-linux-gnu-gcc" AR="aarch64-linux-gnu-ar" RANLIB="aarch64-linux-gnu-ranlib" --disable-wolfcrypt && make
...
examples/pcr/policy.c: In function ‘HexToByte’:
examples/pcr/policy.c:58:18: error: comparison is always false due to limited range of data type [-Werror=type-limits]
   58 |         if ((ch1 < 0) || (ch2 < 0)) {
      |                  ^
examples/pcr/policy.c:58:31: error: comparison is always false due to limited range of data type [-Werror=type-limits]
   58 |         if ((ch1 < 0) || (ch2 < 0)) {
      |                               ^
cc1: all warnings being treated as errors
```